processing/src/main/java/org/apache/druid/math/expr/vector/VectorProcessors.java

```diff
@@ -85,11 +85,14 @@ public static <T> ExprVectorProcessor<T> makeSymmetricalProcessor(
    *
    * @see org.apache.druid.math.expr.ConstantExpr
    */
-  public static <T> ExprVectorProcessor<T> constant(@Nullable String constant, int maxVectorSize)
+  public static <T> ExprVectorProcessor<T> constant(@Nullable Object constant, int maxVectorSize, ExpressionType type)
   {
-    final Object[] strings = new Object[maxVectorSize];
-    Arrays.fill(strings, constant);
-    final ExprEvalObjectVector eval = new ExprEvalObjectVector(strings, ExpressionType.STRING);
+    final Object[] objects = new Object[maxVectorSize];
```

Comment:
we are creating this unnecessarily if type is numeric.

Reply:
numeric types shouldn't ever be calling this, added a defensive check